blog/en/blog/2023/08/11/rate-limit.md

```diff
@@ -0,0 +1,282 @@
+    },
+    "limit-conn": {
+      "conn": 5,
+      "burst": 3,
+      "default_conn_delay": 0.1,
+      "key_type": "var",
+      "key": "remote_addr",
+      "rejected_code": 429,
+      "rejected_msg": "too many concurrent requests!"
+    }
+  }
+}'
+```
+
+In the route, we will just enable the `key-auth` plugin:
+
+```shell
+curl localhost:9180/apisix/admin/routes/rate_limit_consumer -X PUT -d '
+{
+  "uri": "/api",
+  "upstream": {
+    "type": "roundrobin",
+    "nodes": {
+      "upstream:80": 1
+    }
+  },
+  "plugins": {
+    "key-auth": {}
+  }
+}'
+```
+
+We can create a different consumer and set up a different rate limit as well:
+
+```shell
+curl localhost:9180/apisix/admin/consumers -X PUT -d '
+{
+  "username": "bob",
+  "plugins": {
+    "key-auth": {
+      "key": "bobisawesome"
+    },
+    "limit-conn": {
+      "conn": 2,
+      "burst": 1,
+      "default_conn_delay": 0.1,
+      "key_type": "var",
+      "key": "remote_addr",
+      "rejected_code": 429,
+      "rejected_msg": "too many concurrent requests! please upgrade to increase this limit."
+    }
+  }
+}'
+```
+
+Now APISIX will limit the number of concurrent requests these consumers can make independently as desired.
+
+## Grouping Consumers
+
+Configuring each consumer individually can be a tedious process. In addition to supporting third-party authentication providers like [Keycloak](https://apisix.apache.org/docs/apisix/plugins/authz-keycloak/) and [Casdoor](https://apisix.apache.org/docs/apisix/plugins/authz-casdoor/), APISIX has a consumer group object that lets you group individual consumers and apply configurations to the group.
+
+So to apply rate limits to a group of consumers, we can create a consumer group object and add the `limit-count` plugin to it, as shown below:
+
+```shell
+curl localhost:9180/apisix/admin/consumer_groups/team_acme -X PUT -d '
+{
+  "plugins": {
+    "limit-count": {
+      "count": 10,
+      "time_window": 60,
+      "key_type": "var",
+      "key": "remote_addr",
+      "policy": "local",
+      "rejected_code": 429,
+      "group": "team_acme"
+    }
+  }
+}'
+```
+
+We can also create a consumer group for consumers who are from another team on a different plan:
+
+```shell
+curl localhost:9180/apisix/admin/consumer_groups/team_rocket -X PUT -d '
+{
+  "plugins": {
+    "limit-count": {
+      "count": 100,
+      "time_window": 60,
+      "key_type": "var",
+      "key": "remote_addr",
+      "policy": "local",
+      "rejected_code": 429,
+      "group": "team_rocket"
+    }
+  }
+}'
+```
+
+We can add consumers to these groups to attach the rate limit configuration:
+
+```shell
+curl localhost:9180/apisix/admin/consumers -X PUT -d '
+{
+  "username": "eva",
+  "plugins": {
+    "key-auth": {
+      "key": "everydayeva"
+    }
+  },
+  "group_id": "team_acme"
+}'
+```
+
+Now consumers in the `team_acme` group can only send ten requests per minute to your API, while consumers in the `team_rocket` group can send 100.
```

Comment:
Please use a practical example, _e.g._, if both `Alice` and `Bob` belong to team Acme, and `Alice` uses 9 requests in a minute, poor `Bob` has only one request available in this minute